projects/observability/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -498,6 +501,13 @@ export class TableWidgetRendererComponent
     this.columnConfigs$ = this.getColumnConfigs();
   }
 
+  public onVisibleColumnsChange(columns: TableColumnConfig[]): void {
+    this.globalCsvDownloadService.registerDataSource('table-widget-renderer', {
```

Comment:
it is possible that there are multiple table widgets on the same page. we'll need the key to contain some form of ID.

Reply:
done, finally added the model.id